lib/redis_client/cluster.rb

```diff
@@ -93,6 +93,17 @@ def multi(watch: nil, &block)
       ::RedisClient::Cluster::Transaction.new(@router, @command_builder).execute(watch: watch, &block)
     end
 
+    def with(slot_key:, write: true, retry_count: 0, &block)
```

Comment:
I'm wondering the naming of `slot_key` is general for Redis cluster users. Although it may be a bit redundant, would `key_for_slot` be easier to understand?

Reply:
I bounced through a few options with this:

* `key:` - This sounds nice, and sometimes you really will pass a key here (if you're doing MULTI on a single key). But, you can and normally will pass things which aren't even real keys here - i.e. just the bare hashtag like `{foo}`.
* `slot:` - This is what the parameter is _for_ - it calculates the slot based of what you pass in here - but it kind of implies that you actually pass a slot number in here, rather than a key (we could make it work by passing in a slot number too, but i don't see the use case - 99.999% of the time it'll be a hashtag you pass in).
* `hashtag:` - this parameter would be `"foo"` instead of `"{foo}"`. It has the right name, but if you're doing single-key operations you might not actually have a hashtag.
* `slot_key:` - This is where I ended up, it has what it is ("key") and what it's for ("slot") in the name. It's a bit verbose though.
* `key_for_slot:` - This is probably even more clear but also even more verbose than `slot_key`.
* Another option is to make it `slot:`, have it take a number, and also expose a `slot_for(key)` method, so you'd call it as `client.with(slot: client.slot_for(key)) do |conn|`. This feels a bit overly complicated though because i can't think of a use-case for actually passsing a slot number in there that didn't come from hashing a key.

I don't mind too much which one we choose - they all have upsides and downsides. Happy for you to pick something off this list and I'll change the implementation to that :)

Reply:
OR - how about we do _both_ `key:` and `hashtag:`?

* You can pass `key:`, and we'll hash it and use that slot. This will work for single keys _and_ keys with hashtags in them
* You can pass `hashtag:`, and we'll raise an exception if it doesn't match `{...}`. Then, we'll hash it and use that slot like `key`.

So you can write `client.with(hashtag: '{foo}') do |conn|` normally, but if you want to do single key operations, you can do `client.with(key: 'fookey') do |conn|`.  It's slightly redundant because technically `key` can do everything that `hashtag` does, but using `hashtag:` makes it much clearer when reading the code that the block will operate on multiple keys with that hashtag.

Reply:
:+1: I'd like to upvote your idea.